source/Patches/Roles/Modifiers/Modifier.cs

```diff
@@ -65,6 +65,12 @@ public static Modifier GetModifier(PlayerControl player)
                 .FirstOrDefault();
         }
 
+        public static Modifier GetModifier(GameData.PlayerInfo player)
+        {
+            return (from entry in ModifierDictionary where entry.Key == player.PlayerId select entry.Value)
```

Comment:
Possible bug: Do we need a check for `player == null` the way we do in `Role#GetRole(GameData.PlayerInfo)`?

Reply:
not going to lie, i just straight up replicated what i did in Role... and applied it to Modifier, the method in modifier did not do the null check, so i left it out, might be something that needs fixing in multiple places, as a big part of the problem i had with modifiers was that they would be null and stop everything from working, leading to me writing a method for checking if its null which probably isnt needed.

Reply:
The code for `GetRole(GameData.PlayerInfo)` does an explicit check for `player == null` but this code doesn't. This code seems to do something very similar to that code, but it looks *totally* different. Seems like they should be very similar unless there's a compelling reason not to.

Reply:
i have put if (player == null) return null; into both the old function, and the new one i added in modifier.cs (the two listed below)

public static Modifier GetModifier(PlayerControl player)
public static Modifier GetModifier(GameData.PlayerInfo player)